### ISSUE TITLE
Don't let the app crash on 404s

### ISSUE THREAD

**Issue by stefanotorresi** [trento-project/trento#38]
Right now, if I browse to something like `/hosts/foobar` where `foobar` is a not a real host name, the single request will crash and the web engine will return a 500, due to a nil pointer dereference.

here is the log trace:

```
2021/04/28 13:30:03 [Recovery] 2021/04/28 - 13:30:03 panic recovered:
GET /hosts/vmhana01 HTTP/1.1
Host: 51.124.34.113:8080
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9
Accept-Encoding: gzip, deflate
Accept-Language: en,it;q=0.9,en-US;q=0.8,en-GB;q=0.7
Cache-Control: max-age=0
Connection: keep-alive
Referer: http://51.124.34.113:8080/hosts
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.85 Safari/537.36


runtime error: invalid memory address or nil pointer dereference
runtime/panic.go:212 (0x433f1a)
runtime/signal_unix.go:734 (0x44cbd2)
github.com/trento-project/trento/web/hosts.go:213 (0xaf808d)
github.com/gin-gonic/gin@v1.6.3/context.go:161 (0xaf1672)
github.com/trento-project/trento/web/app.go:90 (0xaf1652)
github.com/gin-gonic/gin@v1.6.3/context.go:161 (0xae8290)
github.com/gin-gonic/gin@v1.6.3/recovery.go:83 (0xae8277)
github.com/gin-gonic/gin@v1.6.3/context.go:161 (0xae7313)
github.com/gin-gonic/gin@v1.6.3/logger.go:241 (0xae72d2)
github.com/gin-gonic/gin@v1.6.3/context.go:161 (0xade44f)
github.com/gin-gonic/gin@v1.6.3/gin.go:409 (0xade436)
github.com/gin-gonic/gin@v1.6.3/gin.go:367 (0xaddeec)
github.com/trento-project/trento/web/app.go:86 (0xaf160c)
net/http/server.go:2887 (0x743c82)
net/http/server.go:1952 (0x73f0ac)
runtime/asm_amd64.s:1371 (0x46c9a0)

[GIN] 2021/04/28 - 13:30:03 | 500 |    1.656927ms |     5.94.58.123 | GET      "/hosts/vmhana01"
```

(I just mistakenly hit refresh on a host page after having stopped the agent in that host)

We should handle such invalid requests in a predictable way, and return the appropriate http response (in this case, an error page with a 404 code). Such scenarios also have to be covered by tests.

**Comment by stefanotorresi** [comment]
@fabriziosestito @dirkmueller #93 was a good effort, but the implementation was incomplete; we have recently introduced a dedicated error handler for this type of things:
https://github.com/trento-project/trento/blob/a9140c208fb0f5b6b26032df82dfdfb903d4bfa6/web/errors.go#L5-L33

This allows us to negotiate the content of the response and display a dedicated html error page.

You can see how it is used here:
https://github.com/trento-project/trento/blob/ae7a343ddc3897c458e65fbcc7a6b3baaaec511b/web/environments.go#L113-L156

We also need to apply this fix to _all the handlers_, not just the hosts one! ;)

**Comment by fabriziosestito** [comment]
Resolved in #97
